client/extensions/zoninator/state/data-layer/zones/index.js

```diff
@@ -77,18 +104,61 @@ export const announceZoneCreated = ( { dispatch, getState }, action, response )
 	announceZoneSaved( dispatch, action, response.data );
 };
 
-export const announceFailure = ( { dispatch }, { form } ) => {
+export const announceZoneUpdated = ( { dispatch, getState }, action ) => {
+	const { data, form, siteId, zoneId } = action;
+	const newZone = merge(
+		getZone( getState(), siteId, zoneId ),
+		data,
+	);
+
+	dispatch( initialize( form, data ) );
+	announceZoneSaved( dispatch, action, newZone );
+};
+
+export const announceSaveFailure = ( { dispatch }, { form } ) => {
 	dispatch( stopSubmit( form ) );
 	dispatch( errorNotice(
 		translate( 'There was a problem saving the zone. Please try again.' ),
-		{ id: 'zoninator-zone-create' },
+		{ id: saveZoneNotice },
+	) );
+};
+
+export const deleteZone = ( { dispatch }, action ) => {
+	const { siteId, zoneId } = action;
+
+	dispatch( removeNotice( deleteZoneNotice ) );
+	dispatch( http( {
+		method: 'DELETE',
+		path: `/jetpack-blogs/${ siteId }/rest-api/`,
+		query: {
+			path: `/zoninator/v1/zones/${ zoneId }`,
+		},
+	}, action ) );
+};
+
+export const announceZoneDeleted = ( { dispatch, getState }, { siteId } ) => {
+	page( `${ settingsPath }/${ getSiteSlug( getState(), siteId ) }` );
```

Comment:
Just curious about how this works. Does it redirect and then show the notice on the new page?

Reply:
Yes. It's the same approach as with creating new zones.